packages/wrangler/src/user.tsx

```diff
@@ -662,7 +662,21 @@ async function exchangeRefreshTokenForAccessToken(): Promise<AccessContext> {
     },
   });
   if (response.status >= 400) {
-    throw await response.json();
+    let tokenExhangeResErr = undefined;
+    try {
+      tokenExhangeResErr = await response.json();
+    } catch (e) {
+      // If the response is not JSON, we will attempt to parse it as a string.
+      tokenExhangeResErr = response.text();
```

Comment:
I don't think you can call `.text()` after you've called `.json()` on a `Response` (I'm actually not sure here, but I believe you can't drain a response twice). So what you should probably do is call `.text()` and then `JSON.parse()` on that inside your `try/catch`